apps/passport-client/src/dispatch.ts

```diff
@@ -354,9 +354,10 @@ async function finishAccountCreation(
 
   // Save PCDs to E2EE storage.
   await uploadStorage(user, state.pcds, state.subscriptions);
+  const uploadId = await makeUploadId(state.pcds, state.subscriptions);
```

Comment:
it feels like uploading and generating an upload id should be an atomic operation. there's some chance that edits occur to pcds / subscriptions between these two lines of code. how worried are you about that?

Reply:
Not something I'd thought about before, but yes that's a valid risk.  I guess nothing's really atomic with an `await` in it.  Not the most likely risk at this point right at account creation.  I do realize it's unpredictable when sync() might actually do anything before, after, or during account creation, though.  It can run after setSelf.

I think the cleanest solution to this an other issues, which is already on my list of potential improvements, is to anchor more strongly around the serialized storage blob (in unencrypted form).  That should be what's hashed to get the uploadID, since atomic or not it's what is being uploaded.  It also contains the User object which isn't currently included in the hash.  It should also be what's saved to local storage, rather than multiple separate fields which can get out-of-sync.  I might tackle parts of that in my next diff, but I want to be careful about upsetting too much at once and breaking Zupass during these 3 weeks.